tuner/tuner/libtuner.py

```diff
@@ -221,6 +221,62 @@ def validate_devices(user_devices: list[str]) -> None:
         )
 
 
+def get_valid_benchmark_results(
+    benchmark_results: list[BenchmarkResult],
+) -> list[BenchmarkResult]:
+    """
+    Filter the benchmark_results list to return list with finite `time` values.
+    """
+    filtered_benchmark_results = [r for r in benchmark_results if math.isfinite(r.time)]
+    if len(filtered_benchmark_results) == 0:
+        logging.error("No successful candidate benchmarks.")
+
+    return filtered_benchmark_results
+
+
+def check_baseline_devices_uniqueness(baseline_results: list[BenchmarkResult]) -> bool:
+    seen = set()
+    for result in baseline_results:
+        if result.device_id in seen:
+            return False
+        seen.add(result.device_id)
+    return True
+
+
+def map_baseline_by_device(baseline_results: list[BenchmarkResult]) -> dict[str, float]:
+    return {r.device_id: r.time for r in baseline_results}
+
+
+def detect_baseline_regression(
+    first_baseline_results: list[BenchmarkResult],
+    second_baseline_results: list[BenchmarkResult],
+) -> list[str]:
+    """
+    Detects performance regressions between two sets of baseline results.
+    """
+    regression_device_ids = []
+    first_baseline_by_device = map_baseline_by_device(first_baseline_results)
+    second_baseline_by_device = map_baseline_by_device(second_baseline_results)
+    for device_id in first_baseline_by_device:
+        if device_id not in second_baseline_by_device:
+            continue
+        first_baseline_time = first_baseline_by_device[device_id]
+        second_baseline_time = second_baseline_by_device[device_id]
+
+        if second_baseline_time > first_baseline_time * 1.03:
+            percentage_slower = (
+                (second_baseline_time - first_baseline_time) / first_baseline_time
+            ) * 100
+            logging.warning(
+                f"Performance regression detected on device {device_id}: "
+                f"Baseline time = {first_baseline_time}, Post-baseline time = {second_baseline_time}, "
```

Comment:
```suggestion
                f"First baseline time = {first_baseline_time}, Second baseline time = {second_baseline_time}, "
```

Also, what's the time unit?